src/app/components/Viewer/EBookViewer.jsx

```diff
@@ -14,6 +14,21 @@ class EBookViewer extends React.Component {
 
   componentDidMount() {
     this.parseQueryToState(this.props.location.query.url);
+    this.handleIframeTask = this.handleIframeTask.bind(this);
+    window.addEventListener('message', this.handleIframeTask, false);
+  }
+
+  handleIframeTask(e) {
+    if (e.origin !== 'http://localhost:4444') {
+      return;
+    }
+    if (e.data === 'backButtonClicked') {
+      // TODO: if (previous location within ResearchNow app) {
```

Comment:
This check is not working, and I'm gonna table it for now.  I'm not sure if it's worth it to figure out in react-router-v3 when we're probably gonna upgrade to react-router-4+ in a couple months.

Reply:
What was this attempting to check? Going back to the last link in the app would be a really nice feature.

Reply:
So React-Router's `goBack()` will take you to the last location, but it requires a well set-up history object to be able to check where that is.  Our history object and our router objects are mixed up, I think we're using the `router` object as `history` and `history` is always `undefined`.  I'd love to fix this, but I think part of that fix is going to be upgrading `react-router` because trying to search the internet on info on `react-router@3` is a real pain.